DbAdapterInterface/IField.inl

```diff
@@ -16,5 +15,4 @@ namespace systelab { namespace db {
 	{
 		return !(lhs == rhs);
```

Comment:
For `operator==` like before prefer:
```
return lhs.getType() == rhs.getType()
           && lhs.getName() == rhs.getName();
```
Here is an interface so you need to rely on the virtual getters, but as a general rule of thumb if this was a regular class, go with the member variable directly instead of the getter (one less indirection).

The other thing is that C++ requires you to define the == but the != is implicit if you have the former, so no need for it.

Reply:
> For `operator==` like before prefer:
> 
> ```
> return lhs.getType() == rhs.getType()
>            && lhs.getName() == rhs.getName();
> ```
> 
> Here is an interface so you need to rely on the virtual getters, but as a general rule of thumb if this was a regular class, go with the member variable directly instead of the getter (one less indirection).
> 
> The other thing is that C++ requires you to define the == but the != is implicit if you have the former, so no need for it.

I'll remove it from the class header as well, nice :)